Source/SG/SG.cs

```diff
@@ -23,19 +24,53 @@ public struct InstrumentConfiguration
+            public LocalOscillatorRoutingConfiguration[] LORoutingConfigurations;
+            public LocalOscillatorFrequencyOffsetConfiguration LOOffsetConfiguration;
+
             public static InstrumentConfiguration GetDefault()
             {
                 return new InstrumentConfiguration()
                 {
-                    SelectedPorts="",
+                    SelectedPorts = "",
                     ReferenceClockSource = RfsgFrequencyReferenceSource.PxiClock.ToString(),
                     CarrierFrequency_Hz = 1e9,
-                    DutAverageInputPower_dBm = 0,
+                    DutAverageInputPower_dBm = -10,
                     ExternalAttenuation_dBm = 0,
-                    ShareLOSGToSA = true,
+                    LOSharingMode = LocalOscillatorSharingMode.Automatic,
+                    LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { LocalOscillatorRoutingConfiguration.GetDefault() },
+                    LOOffsetConfiguration = LocalOscillatorFrequencyOffsetConfiguration.GetDefault()
                 };
             }
+
+            public static InstrumentConfiguration GetDefault(NIRfsg sessionHandle)
+            {
+                InstrumentConfiguration instrConfig = GetDefault(); // covers case for sub6 instruments with a single configurable LO
+                // lo routing configurations will now be overridden if the instrument has number of LOs != 1
+                string instrumentModel = sessionHandle.Identity.InstrumentModel;
+                if (instrumentModel.Equals("NI PXIe-5830"))
+                {
+                    instrConfig.SelectedPorts = "IF0";
+                    instrConfig.CarrierFrequency_Hz = 6.5e9;
+                    instrConfig.LORoutingConfigurations[0].ChannelName = "LO2";
+                }
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5831"))
+                {   // these instruments have two configurable LOs
+                    instrConfig.SelectedPorts = "rf0/port0";
+                    instrConfig.CarrierFrequency_Hz = 22.5e9;
+                    LocalOscillatorRoutingConfiguration lo1RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo1RoutingConfig.ChannelName = "LO1";
+                    LocalOscillatorRoutingConfiguration lo2RoutingConfig = LocalOscillatorRoutingConfiguration.GetDefault();
+                    lo2RoutingConfig.ChannelName = "LO2";
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[] { lo1RoutingConfig, lo2RoutingConfig };
+                }
+                else if (Regex.IsMatch(instrumentModel, "NI PXIe-5(82.|645R)")) // matches on any baseband instrument without an LO
+                {
+                    instrConfig.LORoutingConfigurations = new LocalOscillatorRoutingConfiguration[0]; // baseband instruments have no LOs
+                }
+                return instrConfig;

```

Comment:
5645R looks odd to me

Reply:
commented below, but i'll remove references to 5645R. not sure what crossed my mind there since this instrument also clearly has an LO 😏

Reply:
changed this to an else statement

Reply:
I think that this is unnecessary, especially if you incorporate my comments later on in the code. In LabVIEW the only specific behavior I handle is for 5830/31. Otherwise, the default value should be sufficient for any other instrument, because if it is not a supported VST nothing happens anyway. Hence, no reason to set the empty array for the 5820.